### PR TITLE
GraphNG: Make the GraphNG / uPlot flow sync

### PR DESCRIPTION
**Pull request by dprokop**
Yet another alternative to https://github.com/grafana/grafana/pull/32906
Follow up to https://github.com/grafana/grafana/pull/33125 as proposed by @leeoniya 

~This moves data alignment to the TimeSeriesPanel component, making the GraphNG behave as if the alignment happened in PQR.~

~This is quite promising I think as we don't need to:~
~1. Keep the data revision~
~2. Do any tricks with getDerivedStateProps~
~3. Use GraphNGContext for exposing data frame inspection methods and aligned data frame~

~It simplifies the GraphNG lifecycle a bit too. To see what is happening under the hood when you play with TimeSeriesPanel enable debug logs in your console by using `_debug.graphng.enable()`~

- Makes the entire GraphNG/uPlot from sync. 
- Removes async initialisation of uPlot plugins. 
- Removes unnecessary hooks and context.

TODO:
- [x] update tests
- [x] update other panels that use GraphNG
- [x] ContextMenu - fix series color

### CODE REVIEW

File: packages/grafana-ui/src/components/GraphNG/GraphNG.tsx [hunk 1]
```diff
@@ -30,6 +21,7 @@ export interface GraphNGProps extends Themeable {
   width: number;
   height: number;
   data: DataFrame[];
+  alignedData: DataFrame;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/GraphNG/GraphNG.tsx:24]: I keep these two, for now, I guess the `alignedData` should be sufficient, but who knows what the future holds. Tooltip/ContextMenu use the aligned frame, as they can rely on the series index from uPlot, but I have encountered some weirdness when using with PlotLegend.

**Comment by dprokop** [comment]
@leeoniya @torkelo @ryantxu so this PR took a long-awaited turn. The alignment got back to `GraphNG` and I removed all async uPlot hooks registration. This should make it easier to configure uPlot hooks. Consequence of that is GraphNG now exposing config builder and aligned data frame(i.e. for Tooltip to just get the data by index, rather than map to original data frame) via render prop, which seems to be te most readable and understandable way for this scenario.

Entire flow is now sync and free of all side effects we have been encountering before with data vs config updates.

**Comment by leeoniya** [comment]
i have not reviewed everything yet, but generally this feels really good.

i started by running a profile with this vs master, and initially it surfaced some surprises, such as that uPlot's autoranger `minMax()` was running hot during streaming. this was odd for two reasons:

1. it shouldn't run at all since i have both hardMin and hardMax set. it turns out there's no condition to bypass this and set `scale.auto: false`. (addressed by https://github.com/grafana/grafana/pull/33295)
2. why it didnt show up on `master` before this commit is really, really odd. on initial inspection, this PR looks appears to drop fewer frames by staying within the perf budget, so we throttle it less, and thus the renderer is runs hotter. still, the amount of difference here in frames dropped is maybe -20%, but the minMax cost went up by like 200-300%, so :man_shrugging: .

i merged master and tried to simplify plotContext somewhat in https://github.com/grafana/grafana/pull/33294, please take a look and merge that commit into this PR if it's not terrible.

i think i found a bug introduced with this PR for field overrides where the stroke and fill colors for the series were not being set properly (was blue, dark red, etc), not sure why. legend series colors appeared to be properly overridden, though.

File: packages/grafana-ui/src/components/uPlot/plugins/AnnotationsEditorPlugin.tsx [hunk 2]
```diff
@@ -1,59 +1,59 @@
...
+// import useClickAway from 'react-use/lib/useClickAway';
+//
+// interface AnnotationsEditorPluginProps {
+//   onAnnotationCreate: () => void;
+// }
+//
+// /**
+//  * @alpha
+//  */
+// export const AnnotationsEditorPlugin: React.FC<AnnotationsEditorPluginProps> = ({ onAnnotationCreate }) => {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/uPlot/plugins/AnnotationsEditorPlugin.tsx:14]: Commented out for now until we work on the annotations editor

**Review by torkelo — APPROVED** [review]

**Review by zoltanbedi — APPROVED** [review]
Explore looks good.

**Pull request merged**